src/coreComponents/dataRepository/Wrapper.hpp

```diff
@@ -818,100 +777,308 @@ class Wrapper final : public WrapperBase
 
 private:
 
-  /**
-   * @brief Concrete implementation of the packing method.
-   * @tparam DO_PACKING A template parameter to discriminate between actually packing or only computing the packing size.
-   * @param[in,out] buffer The buffer that will receive the packed data.
-   * @param[in] withMetadata Whether to pack string metadata with the underlying data.
-   * @param[in] onDevice Whether to use device-based packing functions
-   *                     (buffer must be either pinned or a device pointer)
-   * @param[out] events A collection of events to poll for completion of async
-   *                    packing kernels ( device packing is incomplete until all
-   *                    events are finalized )
-   * @return The packed size.
-   * @note The @p Impl suffix was used to prevent from a name conflict.
-   */
+  template< typename U = T >
+  typename std::enable_if_t< traits::HasMemorySpaceFunctions< U >, bool >
+  isPackableImpl( ) const
+  {
+    static_assert( std::is_same< T, U >::value, "should only be instantiated for the wrapped type!" );
+    if( reference().getPreviousSpace() == LvArray::MemorySpace::host )
+    {
+      // a type with the mem-space functions, that is not packable on device (e.g. array<string> ), should never be moved to device,
+      //  so it should always have previousSpace == host and checkTouch == true, and we can then host-pack it if possible (should always be
+      // possible)
+      // if the value *is* device-packable, but we have modified it on host, we have to host-pack
+      return bufferOps::is_host_packable< U >;
+    }
+    else
+    {
+      return bufferOps::is_device_packable< decltype( referenceAsView() ) >;
+    }
+  }
+
+  template< typename U = T >
+  typename std::enable_if_t< !traits::HasMemorySpaceFunctions< U >, bool >
+  isPackableImpl( ) const
+  {
+    static_assert( std::is_same< T, U >::value, "should only be instantiated for the wrapped type!" );
+    return bufferOps::is_host_packable< U >;
+  }
```

Comment:
I'm not 100% sure but maybe can you use a `if constexpr ( traits::HasMemorySpaceFunctions< U > ) { ...` instead of using `enable_if_t`?
Maybe we could even remove the need for the extra `U = T` template parameter.
And maybe having an `Impl` version for the `isPackable` function will become useless?